test/e2e/specs/specs-playwright/wp-calypso__coblocks-gutter-spec.js

```diff
@@ -0,0 +1,51 @@
+import assert from 'assert';
+import {
+	setupHooks,
+	DataHelper,
+	LoginFlow,
+	NewPostFlow,
+	GutenbergEditorPage,
+	PricingTableBlock,
+} from '@automattic/calypso-e2e';
+
+describe( DataHelper.createSuiteTitle( 'WPCOM-specific gutter controls' ), () => {
+	let gutenbergEditorPage;
+	let pricingTableBlock;
+	let page;
+
+	setupHooks( ( args ) => {
+		page = args.page;
+	} );
+
+	it( 'Log in', async function () {
+		const loginFlow = new LoginFlow( page, 'gutenbergSimpleSiteUser' );
+		await loginFlow.logIn();
+	} );
+
+	it( 'Start new post', async function () {
+		const newPostFlow = new NewPostFlow( page );
+		await newPostFlow.newPostFromNavbar();
+		gutenbergEditorPage = await GutenbergEditorPage.Expect( page );
+	} );
+
+	it( 'Open settings sidebar', async function () {
+		await gutenbergEditorPage.openSettings();
+	} );
+
+	it( 'Insert Pricing Table block', async function () {
+		const blockHandle = await gutenbergEditorPage.addBlock( 'Pricing Table' );
+		pricingTableBlock = new PricingTableBlock( blockHandle );
+	} );
+
+	it.each`
+		gutterValue
+		${ 'None' }
+		${ 'Small' }
+		${ 'Medium' }
+		${ 'Large' }
+		${ 'Huge' }
```

Comment:
I'd like clarification here when you refer to `prop` - is the suggestion to make a new property in a class, like this?

```
class PricingTableBlock extends ... {
    name: PricingTableBlock
    gutterValues: [
         ....
    ]
```

Reply:
I spent some time trying to do this, but was unable to come to a solution I found satisfactory.

These are the approaches I tried:
1. move the gutter value definition to a static property in the Block file.
2. move the gutter value definition to the type definition file then import that in spec file.

For approach 1, I was unable to have the gutter values defined as a static property in the Block class to be used as the type definition in the argument of `setGutter` method. So there needed to be two locations where this was defined, defeating the purpose.

For approach 2, import of type definition appears to become `undefined` in the spec file. I wonder if it has to do with the JS-TS duality we have in the calypso e2e ecosystem. 

Pinging TS expert @dpasque as he might have some ideas here?

Reply:
Would something like this work in this case?

```ts
const gutterValues = [ 'Small', 'Large' ] as const;
type GutterValues = typeof gutterValues[ number ];

class TheBlock {
    static gutterValues = gutterValues

    setGutter( value: GutterValues ) {
        // set the gutter
    }
}
```

Reply:
Ah, I can see now you kinda did the same thing, although moved the type definition out of the block module. Is it OK if we just leave the definitions contained as in the example above?

Reply:
My 0.02 on these kind of approaches:

- I very much agree with Bart that I think it makes the most sense for these typings to live in the same modules as the block classes they are for (unless they exist more broadly). You can define and export them at the top of the module outside the class.
- If you only need type safety for the types of strings you are passing as an argument, you can just a union string type. I often like to export this type because it helps TypeScript files that consume that type get the right intellisense and be able to use that type for local variables.
```typescript
export type GutterValues = 'Small' | 'Large'
```
- If you want people to be able to interact with an array of the full set of these values, they way you are currently doing it is a great way. In that case though, I would also make sure to export the constant array as well! That way folks can use it and loop on it. This also gives some value to JavaScript file consumers. Types aren't included in transpiled JS, so that's why you were seeing the `undefined`s when you were importing the type. However, you could just import that array and loop over it directly in the test!

Reply:
Thanks for input, I had thought the best practice was to group all type definitions in one file but it doesn't seem that's the case.

I've made the following changes:

1. move the type and array definition of the gutter values to the PricingTableBlock.
2. assign a static property `gutterValues` to the PricingTableBlock class using the array defined in 1.
3. use the static property `gutterValues` to build the parametrized test case in the spec.